Configuration/Test/test.py

```diff
@@ -0,0 +1,534 @@
+process.o1 = cms.OutputModule("PoolOutputModule",
+    outputCommands = cms.untracked.vstring('keep *'),
+    fileName = cms.untracked.string('file:test_1k.root')
+)
+process.outpath = cms.EndPath(process.o1)
+
+################## STEP 1 - process.generator
+process.source = cms.Source("EmptySource")
+
+# Use random number generator service
+#process.load("Configuration.TotemCommon.RandomNumbers_cfi")
+
+# particle generator paramteres
+process.load("Configuration.StandardSequences.Services_cff")
+
+process.load("Configuration.Generator.FlatLogKsiLogT_pp_13TeV_cfi")
+process.RandomNumberGeneratorService = cms.Service("RandomNumberGeneratorService",
+    g4SimHits = cms.PSet(initialSeed = cms.untracked.uint32(9876)),
+    SimG4Object = cms.PSet(initialSeed =cms.untracked.uint32(9876)),
+    RPSiDetDigitizer = cms.PSet(initialSeed =cms.untracked.uint32(137137)),
+    sourceSeed = cms.PSet(initialSeed =cms.untracked.uint32(98765)),
+    generator = cms.PSet(initialSeed = cms.untracked.uint32(98766)),
+    SmearingGenerator = cms.PSet(initialSeed =cms.untracked.uint32(3849)),
+    T2Digis = cms.PSet(initialSeed =cms.untracked.uint32(98765)),
+    T2MCl = cms.PSet(initialSeed =cms.untracked.uint32(24141)),
+    RPFastStationSimulation = cms.PSet(initialSeed =cms.untracked.uint32(12)),
+    RPFastFullSimulation = cms.PSet(initialSeed =cms.untracked.uint32(13)),
+    mix = cms.PSet(initialSeed = cms.untracked.uint32(24141)),
+    LHCTransport = cms.PSet(initialSeed = cms.untracked.uint32(24143), engineName = cms.untracked.string('TRandom3')
+  )
+
+)
+################# STEP 2 process.SmearingGenerator
+process.SmearingGenerator = cms.EDProducer("GaussEvtVtxEnergyGenerator",
+    src   = cms.InputTag("generator"),
+    MeanX = cms.double(0.0),
+    MeanY = cms.double(0.0),
+    MeanZ = cms.double(0.0),
+    SigmaX = cms.double(0.000001),
+    SigmaY = cms.double(0.000001),
+    SigmaZ = cms.double(0.000001),
+    TimeOffset = cms.double(0.0)
+)
+# declare optics parameters
+#process.load("Configuration.TotemOpticsConfiguration.OpticsConfig_6500GeV_0p8_145urad_cfi")
+process.BeamOpticsParamsESSource = cms.ESSource("BeamOpticsParamsESSource",
+    BeamEnergy = cms.double(6500.0), # Gev
+    ProtonMass = cms.double(0.938272029), # Gev
+    LightSpeed = cms.double(300000000.0),
+    NormalizedEmittanceX = cms.double(3.75e-06),
+    NormalizedEmittanceY = cms.double(3.75e-06),
+    BetaStarX = cms.double(0.4), # m
+    BetaStarY = cms.double(0.4), # m
+    CrossingAngleX = cms.double(185e-6),
+    CrossingAngleY = cms.double(0.0),
+    BeamDisplacementX = cms.double(0.0), # m
+    BeamDisplacementY = cms.double(0.0), # m
+    BeamDisplacementZ = cms.double(0.0), # m
+    BunchSizeZ = cms.double(0.07), # m
+    MeanXi = cms.double(0.0), # energy smearing
+    SigmaXi = cms.double(0.0001)
+)
+
+process.ProtonTransportFunctionsESSource = cms.ESProducer("ProtonTransportFunctionsESSource",
+    opticsFile = cms.string(''), # automatic
```

Comment:
is `opticsFile` field filled later at some point ?

Reply:
Znalazłem taki kod: https://github.com/TOTEM-CERN/cmssw/blob/simulation_8_1_0-0p4/TotemCondFormats/ProtonTransportFunctions/plugins/ProtonTransportFunctionsESSource.cc#L94